tests/oclinter/run.sh

```diff
@@ -18,15 +18,19 @@ FAIL=0
 
 export PLUGIN_DIR=$(/usr/bin/env python -c \
       'import openconfig_pyang; import os; \
-        print "%s/plugins" % \
-          os.path.dirname(openconfig_pyang.__file__)')
+       print("{}/plugins".format(os.path.dirname(openconfig_pyang.__file__)))')
 
 for i in `find $TESTDIR -mindepth 1 -maxdepth 1 -type d`; do
   if [ -e $i/Makefile ]; then
     failed=0
 
-    okres=$(cd $i; make ok &>/dev/null; echo $?)
-    borkres=$(cd $i; make broken &>/dev/null; echo $?)
+    cd $i
+    make ok
+    okres=$(echo $?)
+
+    cd $i
```

Comment:
Does this work -- since in the `okres` and `borkres` case above, we're executing in a subshell, doesn't the second `cd $i` essentially resolve to `cd $i; cd $i`?

Reply:
Didn't quite get what you meant here. Do you mean the second `cd $i` is redundant? Anyhow I've restored the old subshell behaviour and removed the `>/dev/null` to allow for debugging, which was the original purpose of this change.

Reply:
I mean that the change you made doesn't result in equivalent behaviour.

If we have this directory structure:

```
[12:47] ~/tmp                                                                 ⍉
▶ find .
.
./a
./test.sh
./b
```

and in `test.sh` you do:

```bash
ok=$(cd a; echo $?)
nok=$(cd b; echo $?)

echo $ok
echo $nok
```

then this says "start a new shell, and `cd` into directory `a` and give the return code", then it says  "start another new shell and cd into directory `b` and give the return code`. This means we `cd` into `./a` and `./b`.

In the case that you've written above - then you never start these new shells, so we don't discard the context. The equivalent is:

```
cd a
ok=$(echo $?)
cd b
nok=$(echo $?)

echo $ok
echo $nok
```

In this case, we don't `cd` into `./a` and `./b`, but rather we try `./a` and then `./a/b` (fully resolved). In my example above, `nok` will return 0 in the first case (with subshells) and 1 in the second case, because `./a/b` doesn't exist.

In this script, you'll likely just get an error because `./$i/$i` likely doesn't exist. If it does, you'll run the `make broken` command in the wrong directory.

Reply:
I think the best thing to do here is to keep the subshells and just remove the `>/dev/null` if that's the intention of the change.

Reply:
Sorry I missed that, thanks for the explanation!